guidance/llms/_openai.py

```diff
@@ -731,13 +731,14 @@ def __init__(self, model=None, client_id=None, authority=None, caching=True, max
         if os.path.exists(self._token_cache_path):
             self._token_cache.deserialize(open(self._token_cache_path, 'r').read())
 
-        self._rest_headers["X-ModelType"] = self.model_name
+        if( rest_call ):
+            self._rest_headers["X-ModelType"] = self.model_name
 
     @property
-    def token(self):
+    def api_key(self):
```

Comment:
These properties aren't used anywhere so this should all be fine, but just for my own understanding -- any particular reason for the rename to api_key?

Reply:
When the parent class OpenAI makes a call it passes the value of the api_key property to the server.  MLASOpenAI overrides the api_key property with the AAD token.  I think the OpenAI class previously looked at the value of 'token', but when that changed, the MLASOpenAI functionality broke.

<img width="578" alt="image" src="https://github.com/microsoft/guidance/assets/5982160/4b30f946-a779-4042-9c2e-6072d11e2aa1">

Reply:
Ah got it, now the fix makes perfect sense. Thanks for clarifying :)!